From c2c98ddde2665d12e34f17c4eac90832df720114 Mon Sep 17 00:00:00 2001
From: Tomas Orsava <tomas.n@orsava.cz>
Date: Thu, 16 Jun 2016 18:56:18 +0200
Subject: [PATCH] Raise an error when STARTTLS fails

Based on an upstream change by Benjamin Peterson <benjamin@python.org>
- in changeset 101886:b3ce713fb9be 2.7
- https://hg.python.org/cpython/rev/b3ce713fb9be
---
 Lib/smtplib.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Lib/smtplib.py b/Lib/smtplib.py
index 8388b98..e1651c0 100755
--- a/Lib/smtplib.py
+++ b/Lib/smtplib.py
@@ -656,6 +656,11 @@ class SMTP:
             self.ehlo_resp = None
             self.esmtp_features = {}
             self.does_esmtp = 0
+        else:
+            # RFC 3207:
+            # 501 Syntax error (no parameters allowed)
+            # 454 TLS not available due to temporary reason
+            raise SMTPResponseException(resp, reply)
         return (resp, reply)
 
     def sendmail(self, from_addr, to_addrs, msg, mail_options=[],
-- 
2.5.5

